### PR TITLE
[WFLY-11360] Revert previous take on exposing TransationSynchronizationRegistry via a capability using ContextTransactionSynchronizationRegistry; instead complete that part of WFLY-11166 using a standard  Service-installing capability

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.jboss.org/browse/WFLY-11360
https://issues.jboss.org/browse/WFLY-11166

### CODE REVIEW

**Comment by bstansberry** [comment]
@pferraro 1) Does this look ok? See WFLY-11360 for why what I did before re ContextTransactionSynchronizationRegistry is no good.  2) Do you have any sense the failures in "Linux" and "Windows" CI jobs are related?  I see lots of clustering failures lately.

**Comment by pferraro** [comment]
@bstansberry I'll have a look in a minute.
Those failure are unrelated and are due to https://issues.jboss.org/browse/ISPN-9701, for which an upstream fix should arrive shortly.  We were previously inoculated against this by lifecycle locking in the distributed session manager which was dropped by https://github.com/wildfly/wildfly/commit/88f0cd5bd076ebbbf6fa501c8afa294a7706b063

**Review by pferraro — CHANGES_REQUESTED** [review]
This is technically fine, although I've suggested some tweaks.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionServiceConfigurator.java [hunk 1]
```diff
@@ -57,6 +61,8 @@
     private volatile long timeout;
     private volatile TransactionMode mode;
     private volatile Dependency transactionDependency;
+    private volatile ServiceName tsrServiceName;
+    private volatile Supplier<TransactionSynchronizationRegistry> tsrSupplier;
```
[the remark this answers is not in the document]
Comment (bstansberry) [reply clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionServiceConfigurator.java]: Thanks, Paul; I'll make all these changes.

**Comment by scottmarlow** [comment]
I'm not sure of a quick/easy way to add a unit test, so will try building/manually testing the current pr (will repeat after code changes)

**Comment by scottmarlow** [comment]
JCAOrderedLastSynchronizationList#registerInterposedSynchronization was called to register the JPA container (org.jboss.as.jpa.transaction.TransactionUtil$SessionSynchronization) sync and JCA (org.jboss.jca.core.connectionmanager.transaction.TransactionSynchronizer) sync, which means this PR addresses the  WFLY-11360 regression.

**Comment by bstansberry** [comment]
@scottmarlow Thanks!  I just updated this to reflect incorporate the inputs from @pferraro.  That shouldn't change any behavior though; it was all in one infinispan extension class and was just a change to use a couple helper classes in place of some handwritten code.

**Review by pferraro — APPROVED** [review]
Clustering changes approved.

**Comment by kabir** [comment]
@maeste Can you approve the jca changes?

**Review by maeste — APPROVED** [review]

**Pull request merged**